src/idl_gen_cpp.cpp

```diff
@@ -2522,6 +2525,16 @@ class CppGenerator : public BaseGenerator {
                 ? ".type"
                 : (field.value.type.element == BASE_TYPE_UNION ? ".value" : "");
         code += "{ _o->" + name + ".resize(_e->size()); ";
+        if (!vector_of_bool && !field.value.type.enum_def &&
+            IsOneByte(field.value.type.element)) {
+          // For vectors of bytes, memcpy is used to improve performance.
+          // This doesn't work for:
+          //  - std::vector<bool> because of the template specialization.
+          //  - enum types because they have to be explicitly static_cast.
+          //  - multiple-byte types due to endianness.
+          code += "memcpy(_o->" + name + ".data(), _e->data(), _e->size()); }";
```

Comment:
Could we just use std::copy (or std::copy_n) instead of memcpy as well, as it is more C++?

Here is one opinion on the matter: https://stackoverflow.com/questions/4707012/is-it-better-to-use-stdmemcpy-or-stdcopy-in-terms-to-performance

Reply:
yes this is a better idea. Thanks!

done.